// tdesign-vue-next/cjs/date-picker/base/Footer.js
/**
 * tdesign v1.11.1
 * (c) 2025 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var _defineProperty = require('@babel/runtime/helpers/defineProperty');
var Vue = require('vue');
var hooks_useConfig = require('../../hooks/useConfig.js');
var button_index = require('../../button/index.js');
var configProvider_useConfig = require('../../config-provider/useConfig.js');
require('../../_chunks/dep-bc99882c.js');
require('../../_chunks/dep-dbac77fd.js');
require('@babel/runtime/helpers/typeof');
require('../../_chunks/dep-c0c71352.js');
require('dayjs');
require('../../_chunks/dep-9e37eebf.js');
require('../../_chunks/dep-a01b02fa.js');
require('../../_chunks/dep-7b70fa2b.js');
require('../../_chunks/dep-901ff2b1.js');
require('../../_chunks/dep-f3362596.js');
require('../../_chunks/dep-8af7b432.js');
require('../../_chunks/dep-ae77d744.js');
require('../../_chunks/dep-8db04cd4.js');
require('../../_chunks/dep-0d790559.js');
require('../../_chunks/dep-750b6c90.js');
require('../../_chunks/dep-fd23ce18.js');
require('../../_chunks/dep-ec56dcd9.js');
require('../../_chunks/dep-b402ceae.js');
require('../../_chunks/dep-0d817d77.js');
require('../../_chunks/dep-bbc5435a.js');
require('../../_chunks/dep-10ffbad0.js');
require('../../_chunks/dep-89db45fd.js');
require('../../_chunks/dep-b75e3209.js');
require('../../_chunks/dep-55ae3cb5.js');
require('../../_chunks/dep-481d05e9.js');
require('../../_chunks/dep-b8cfc98c.js');
require('../../_chunks/dep-dc345794.js');
require('../../_chunks/dep-acdd3481.js');
require('../../button/button.js');
require('@babel/runtime/helpers/toConsumableArray');
require('../../loading/index.js');
require('../../_chunks/dep-f6db9929.js');
require('../../loading/plugin.js');
require('../../_chunks/dep-ade5be71.js');
require('@babel/runtime/helpers/slicedToArray');
require('../../loading/icon/gradient.js');
require('../../_chunks/dep-dca6be9c.js');
require('../../_chunks/dep-623661fb.js');
require('@babel/runtime/helpers/objectWithoutProperties');
require('../../_chunks/dep-14f725de.js');
require('../../_chunks/dep-31fe118c.js');
require('../../_chunks/dep-db2cd3d2.js');
require('../../utils/dom.js');
require('../../utils/render-tnode.js');
require('../../_chunks/dep-9a234926.js');
require('../../_chunks/dep-179c4586.js');
require('../../_chunks/dep-65ed94ad.js');
require('../../_chunks/dep-93deee44.js');
require('../../_chunks/dep-b74aef17.js');
require('../../_chunks/dep-53d908b3.js');
require('../../_chunks/dep-172ba234.js');
require('../../_chunks/dep-2930dcc3.js');
require('../../loading/props.js');
require('../../hooks/useTeleport.js');
require('../../_chunks/dep-a246b8d5.js');
require('../../_chunks/dep-d9378865.js');
require('../../_chunks/dep-3a398159.js');
require('../../_chunks/dep-470ac849.js');
require('../../_chunks/dep-973f44c6.js');
require('../../_chunks/dep-b377c7d2.js');
require('../../_chunks/dep-029e312d.js');
require('../../utils/withInstall.js');
require('../../button/props.js');
require('../../hooks/useRipple.js');
require('../../hooks/useKeepAnimation.js');
require('../../hooks/tnode.js');
require('../../hooks/useDisabled.js');
require('../../_chunks/dep-5c1c1647.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var _defineProperty__default = /*#__PURE__*/_interopDefaultLegacy(_defineProperty);

function _isSlot(s) {
  return typeof s === 'function' || Object.prototype.toString.call(s) === '[object Object]' && !Vue.isVNode(s);
}
var TDateFooter = Vue.defineComponent({
  name: "TDatePickerTable",
  props: {
    enableTimePicker: Boolean,
    presetsPlacement: String,
    presets: Object,
    needConfirm: {
      type: Boolean,
      "default": true
    },
    selectedValue: [String, Number, Array, Date],
    onPresetClick: Function,
    onConfirmClick: Function
  },
  setup: function setup(props) {
    var COMPONENT_NAME = hooks_useConfig.usePrefixClass("date-picker__footer");
    var presetsClass = hooks_useConfig.usePrefixClass("date-picker__presets");
    var _useConfig = configProvider_useConfig.useConfig("datePicker"),
      t = _useConfig.t,
      globalConfig = _useConfig.globalConfig;
    var footerClass = Vue.computed(function () {
      return [COMPONENT_NAME.value, "".concat(COMPONENT_NAME.value, "--").concat(props.presetsPlacement)];
    });
    return function () {
      var _slot;
      return Vue.createVNode("div", {
        "class": footerClass.value
      }, [Vue.createVNode("div", {
        "class": presetsClass.value
      }, [props.presets && Object.keys(props.presets).map(function (key) {
        return Vue.createVNode(button_index.Button, {
          "key": key,
          "size": "small",
          "variant": "text",
          "onClick": function onClick(e) {
            var _props$onPresetClick;
            return (_props$onPresetClick = props.onPresetClick) === null || _props$onPresetClick === void 0 ? void 0 : _props$onPresetClick.call(props, props.presets[key], {
              e: e,
              preset: _defineProperty__default["default"]({}, key, props.presets[key])
            });
          }
        }, _isSlot(key) ? key : {
          "default": function _default() {
            return [key];
          }
        });
      })]), props.enableTimePicker && props.needConfirm && Vue.createVNode(button_index.Button, {
        "disabled": !props.selectedValue,
        "size": "small",
        "theme": "primary",
        "onClick": function onClick(e) {
          var _props$onConfirmClick;
          return (_props$onConfirmClick = props.onConfirmClick) === null || _props$onConfirmClick === void 0 ? void 0 : _props$onConfirmClick.call(props, {
            e: e
          });
        }
      }, _isSlot(_slot = t(globalConfig.value.confirm)) ? _slot : {
        "default": function _default() {
          return [_slot];
        }
      })]);
    };
  }
});

exports["default"] = TDateFooter;
