// tdesign-vue-next/lib/date-picker/panel/ExtraContent.js
/**
 * tdesign v1.11.1
 * (c) 2025 tdesign
 * @license MIT
 */

import { defineComponent, computed, createVNode } from 'vue';
import TDateFooter from '../base/Footer.js';
import '@babel/runtime/helpers/defineProperty';
import '../../hooks/useConfig.js';
import '../../config-provider/useConfig.js';
import 'lodash-es';
import '../../config-provider/context.js';
import '../../_chunks/dep-a46dd404.js';
import '@babel/runtime/helpers/typeof';
import '../../_chunks/dep-6d42da9c.js';
import 'dayjs';
import '../../button/index.js';
import '../../button/button.js';
import '@babel/runtime/helpers/toConsumableArray';
import '../../loading/index.js';
import '../../loading/directive.js';
import '../../loading/plugin.js';
import '../../_chunks/dep-07bb7e0e.js';
import '@babel/runtime/helpers/slicedToArray';
import '../../loading/icon/gradient.js';
import '../../_chunks/dep-7b6ca170.js';
import '../../_chunks/dep-9fb67ff6.js';
import '@babel/runtime/helpers/objectWithoutProperties';
import '../../utils/dom.js';
import '../../utils/render-tnode.js';
import '../../loading/props.js';
import '../../hooks/useTeleport.js';
import '../../utils/withInstall.js';
import '../../button/props.js';
import '../../hooks/useRipple.js';
import '../../hooks/useKeepAnimation.js';
import '../../hooks/tnode.js';
import '../../hooks/useDisabled.js';

var TExtraContent = defineComponent({
  name: "TExtraContent",
  props: {
    presets: Object,
    enableTimePicker: Boolean,
    presetsPlacement: String,
    needConfirm: {
      type: Boolean,
      "default": true
    },
    onPresetClick: Function,
    onConfirmClick: Function,
    selectedValue: [String, Number, Array, Date]
  },
  setup: function setup(props) {
    var showPanelFooter = computed(function () {
      return props.enableTimePicker && props.needConfirm || props.presets;
    });
    return function () {
      return showPanelFooter.value ? createVNode(TDateFooter, {
        "presets": props.presets,
        "onPresetClick": props.onPresetClick,
        "enableTimePicker": props.enableTimePicker,
        "onConfirmClick": props.onConfirmClick,
        "presetsPlacement": props.presetsPlacement,
        "selectedValue": props.selectedValue,
        "needConfirm": props.needConfirm
      }, null) : null;
    };
  }
});

export { TExtraContent as default };
