// tdesign-vue-next/cjs/select-input/useSingle.js
/**
 * tdesign v1.11.1
 * (c) 2025 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var _objectWithoutProperties = require('@babel/runtime/helpers/objectWithoutProperties');
var _slicedToArray = require('@babel/runtime/helpers/slicedToArray');
var _defineProperty = require('@babel/runtime/helpers/defineProperty');
var Vue = require('vue');
var input_index = require('../input/index.js');
var loading_index = require('../loading/index.js');
var hooks_tnode = require('../hooks/tnode.js');
var hooks_useConfig = require('../hooks/useConfig.js');
var hooks_useDefaultValue = require('../hooks/useDefaultValue.js');
var hooks_useDisabled = require('../hooks/useDisabled.js');
var hooks_useReadonly = require('../hooks/useReadonly.js');
var pick = require('../_chunks/dep-eee7af37.js');
var isObject = require('../_chunks/dep-ae77d744.js');
require('../input/input.js');
require('tdesign-icons-vue-next');
require('../input/props.js');
require('../hooks/useGlobalIcon.js');
require('../config-provider/useConfig.js');
require('../_chunks/dep-bc99882c.js');
require('../_chunks/dep-dbac77fd.js');
require('@babel/runtime/helpers/typeof');
require('../_chunks/dep-c0c71352.js');
require('dayjs');
require('../_chunks/dep-9e37eebf.js');
require('../_chunks/dep-a01b02fa.js');
require('../_chunks/dep-7b70fa2b.js');
require('../_chunks/dep-901ff2b1.js');
require('../_chunks/dep-f3362596.js');
require('../_chunks/dep-8af7b432.js');
require('../_chunks/dep-8db04cd4.js');
require('../_chunks/dep-0d790559.js');
require('../_chunks/dep-750b6c90.js');
require('../_chunks/dep-fd23ce18.js');
require('../_chunks/dep-ec56dcd9.js');
require('../_chunks/dep-b402ceae.js');
require('../_chunks/dep-0d817d77.js');
require('../_chunks/dep-bbc5435a.js');
require('../_chunks/dep-10ffbad0.js');
require('../_chunks/dep-89db45fd.js');
require('../_chunks/dep-b75e3209.js');
require('../_chunks/dep-55ae3cb5.js');
require('../_chunks/dep-481d05e9.js');
require('../_chunks/dep-b8cfc98c.js');
require('../_chunks/dep-dc345794.js');
require('../_chunks/dep-acdd3481.js');
require('../input/useInput.js');
require('../form/const.js');
require('../hooks/useKeepAnimation.js');
require('../hooks/useVModel.js');
require('../_chunks/dep-172ba234.js');
require('../_chunks/dep-53d908b3.js');
require('../_chunks/dep-179c4586.js');
require('../_chunks/dep-65ed94ad.js');
require('../input/useLengthLimit.js');
require('../_chunks/dep-02147c0b.js');
require('../_chunks/dep-623661fb.js');
require('@babel/runtime/helpers/toConsumableArray');
require('../_chunks/dep-14f725de.js');
require('../_chunks/dep-31fe118c.js');
require('../_chunks/dep-db2cd3d2.js');
require('../_chunks/dep-5c1c1647.js');
require('../input/useInputEventHandler.js');
require('../input/useInputWidth.js');
require('../hooks/useResizeObserver.js');
require('../utils/render-tnode.js');
require('../_chunks/dep-9a234926.js');
require('../_chunks/dep-93deee44.js');
require('../_chunks/dep-b74aef17.js');
require('../_chunks/dep-2930dcc3.js');
require('../input/input-group.js');
require('../input/input-group-props.js');
require('../utils/withInstall.js');
require('../_chunks/dep-f6db9929.js');
require('../loading/plugin.js');
require('../_chunks/dep-ade5be71.js');
require('../loading/icon/gradient.js');
require('../_chunks/dep-dca6be9c.js');
require('../utils/dom.js');
require('../loading/props.js');
require('../hooks/useTeleport.js');
require('../_chunks/dep-a246b8d5.js');
require('../_chunks/dep-d9378865.js');
require('../_chunks/dep-3a398159.js');
require('../_chunks/dep-470ac849.js');
require('../_chunks/dep-973f44c6.js');
require('../_chunks/dep-b377c7d2.js');
require('../_chunks/dep-029e312d.js');
require('../_chunks/dep-b694cee7.js');
require('../_chunks/dep-438edd07.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var _objectWithoutProperties__default = /*#__PURE__*/_interopDefaultLegacy(_objectWithoutProperties);
var _slicedToArray__default = /*#__PURE__*/_interopDefaultLegacy(_slicedToArray);
var _defineProperty__default = /*#__PURE__*/_interopDefaultLegacy(_defineProperty);

var _excluded = ["tips"];
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty__default["default"](e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var COMMON_PROPERTIES = ["status", "clearable", "disabled", "label", "placeholder", "readonly", "prefixIcon", "suffix", "suffixIcon", "onPaste", "onMouseenter", "onMouseleave", "size", "autofocus"];
var DEFAULT_KEYS = {
  label: "label",
  value: "value",
  children: "children"
};
function getInputValue(value, keys) {
  var iKeys = _objectSpread(_objectSpread({}, DEFAULT_KEYS), keys);
  return isObject.isObject(value) ? value[iKeys.label] : value;
}
function useSingle(props, context, popupRef) {
  var _props$defaultInputVa;
  var _toRefs = Vue.toRefs(props),
    value = _toRefs.value,
    keys = _toRefs.keys,
    propsInputValue = _toRefs.inputValue;
  var classPrefix = hooks_useConfig.usePrefixClass();
  var isSingleFocus = Vue.ref(props.autofocus);
  var inputRef = Vue.ref();
  var _useDefaultValue = hooks_useDefaultValue["default"](propsInputValue, (_props$defaultInputVa = props.defaultInputValue) !== null && _props$defaultInputVa !== void 0 ? _props$defaultInputVa : "", props.onInputChange, "inputValue"),
    _useDefaultValue2 = _slicedToArray__default["default"](_useDefaultValue, 2),
    inputValue = _useDefaultValue2[0],
    setInputValue = _useDefaultValue2[1];
  var renderTNode = hooks_tnode.useTNodeJSX();
  var disable = hooks_useDisabled.useDisabled();
  var isReadonly = hooks_useReadonly.useReadonly();
  var commonInputProps = Vue.computed(function () {
    return _objectSpread(_objectSpread({}, pick.pick(props, COMMON_PROPERTIES)), {}, {
      disabled: disable.value,
      readonly: isReadonly.value
    });
  });
  var onInnerClear = function onInnerClear(context2) {
    var _context2$e, _props$onClear;
    context2 === null || context2 === void 0 || (_context2$e = context2.e) === null || _context2$e === void 0 || _context2$e.stopPropagation();
    (_props$onClear = props.onClear) === null || _props$onClear === void 0 || _props$onClear.call(props, context2);
    setInputValue("", {
      trigger: "clear"
    });
  };
  var onInnerInputChange = function onInnerInputChange(value2, context2) {
    if (props.allowInput) {
      setInputValue(value2, _objectSpread(_objectSpread({}, context2), {}, {
        trigger: context2.trigger || "input"
      }));
    }
  };
  var renderSelectSingle = function renderSelectSingle(popupVisible) {
    var _context$attrs;
    var singleValueDisplay = renderTNode("valueDisplay");
    var displayedValue = popupVisible && props.allowInput ? inputValue.value : getInputValue(value.value, keys.value);
    var prefixContent = renderPrefixContent(singleValueDisplay, popupVisible);
    var inputProps = _objectSpread(_objectSpread({}, commonInputProps.value), {}, {
      value: renderInputDisplay(singleValueDisplay, displayedValue, popupVisible),
      label: prefixContent.length ? function () {
        return prefixContent;
      } : void 0,
      autoWidth: props.autoWidth,
      readonly: !props.allowInput || isReadonly.value,
      placeholder: renderPlaceholder(singleValueDisplay),
      suffixIcon: !disable.value && props.loading ? function () {
        return Vue.createVNode(loading_index.Loading, {
          "loading": true,
          "size": "small"
        }, null);
      } : props.suffixIcon,
      showClearIconOnEmpty: Boolean(props.clearable && (inputValue.value || displayedValue) && !disable.value && !isReadonly.value)
    }, props.inputProps);
    var _context$slots = context.slots;
      _context$slots.tips;
      var slots = _objectWithoutProperties__default["default"](_context$slots, _excluded);
    var inputClassProps = popupVisible ? ["".concat(classPrefix.value, "-input--focused"), "".concat(classPrefix.value, "-is-focused"), inputProps === null || inputProps === void 0 ? void 0 : inputProps.inputClass] : inputProps === null || inputProps === void 0 ? void 0 : inputProps.inputClass;
    var onEnter = function onEnter(val, context2) {
      var _props$onEnter;
      (_props$onEnter = props.onEnter) === null || _props$onEnter === void 0 || _props$onEnter.call(props, value.value, _objectSpread(_objectSpread({}, context2), {}, {
        inputValue: val
      }));
    };
    var onFocus = function onFocus(val, context2) {
      var _popupRef$value, _props$onFocus;
      var overlayState = (_popupRef$value = popupRef.value) === null || _popupRef$value === void 0 ? void 0 : _popupRef$value.getOverlayState();
      if (isSingleFocus.value || overlayState !== null && overlayState !== void 0 && overlayState.hover) return;
      isSingleFocus.value = true;
      (_props$onFocus = props.onFocus) === null || _props$onFocus === void 0 || _props$onFocus.call(props, value.value, _objectSpread(_objectSpread({}, context2), {}, {
        inputValue: val
      }));
    };
    var onBlur = function onBlur(val, context2) {
      var _popupRef$value2, _props$onBlur;
      var overlayState = (_popupRef$value2 = popupRef.value) === null || _popupRef$value2 === void 0 ? void 0 : _popupRef$value2.getOverlayState();
      isSingleFocus.value = false;
      if (overlayState !== null && overlayState !== void 0 && overlayState.hover) return;
      (_props$onBlur = props.onBlur) === null || _props$onBlur === void 0 || _props$onBlur.call(props, value.value, _objectSpread(_objectSpread({}, context2), {}, {
        inputValue: val
      }));
    };
    return Vue.createVNode(input_index.Input, Vue.mergeProps({
      "ref": inputRef,
      "style": (_context$attrs = context.attrs) === null || _context$attrs === void 0 ? void 0 : _context$attrs.style
    }, _objectSpread({
      onChange: onInnerInputChange,
      onClear: onInnerClear,
      onEnter: onEnter,
      onFocus: onFocus,
      onBlur: onBlur
    }, inputProps), {
      "inputClass": inputClassProps
    }), slots);
  };
  var renderPrefixContent = function renderPrefixContent(singleValueDisplay, popupVisible) {
    var label = renderTNode("label");
    if (!label && !singleValueDisplay) {
      return [];
    }
    if (singleValueDisplay) {
      var _props$valueDisplayOp, _props$valueDisplayOp2;
      if ((_props$valueDisplayOp = props.valueDisplayOptions) !== null && _props$valueDisplayOp !== void 0 && _props$valueDisplayOp.usePlaceholder && !value.value || (_props$valueDisplayOp2 = props.valueDisplayOptions) !== null && _props$valueDisplayOp2 !== void 0 && _props$valueDisplayOp2.useInputDisplay && popupVisible) {
        return [label];
      }
    }
    return [label, singleValueDisplay];
  };
  var renderInputDisplay = function renderInputDisplay(singleValueDisplay, displayedValue, popupVisible) {
    if (singleValueDisplay) {
      var _props$valueDisplayOp3, _props$valueDisplayOp4;
      if (!((_props$valueDisplayOp3 = props.valueDisplayOptions) !== null && _props$valueDisplayOp3 !== void 0 && _props$valueDisplayOp3.useInputDisplay) || (_props$valueDisplayOp4 = props.valueDisplayOptions) !== null && _props$valueDisplayOp4 !== void 0 && _props$valueDisplayOp4.useInputDisplay && !popupVisible) {
        return void 0;
      }
    }
    return displayedValue;
  };
  var renderPlaceholder = function renderPlaceholder(singleValueDisplay) {
    if (singleValueDisplay) {
      var _props$valueDisplayOp5, _props$valueDisplayOp6;
      if (!((_props$valueDisplayOp5 = props.valueDisplayOptions) !== null && _props$valueDisplayOp5 !== void 0 && _props$valueDisplayOp5.usePlaceholder) || (_props$valueDisplayOp6 = props.valueDisplayOptions) !== null && _props$valueDisplayOp6 !== void 0 && _props$valueDisplayOp6.usePlaceholder && value.value) {
        return "";
      }
    }
    return props.placeholder;
  };
  return {
    inputRef: inputRef,
    isSingleFocus: isSingleFocus,
    commonInputProps: commonInputProps,
    singleInputValue: inputValue,
    onInnerClear: onInnerClear,
    renderSelectSingle: renderSelectSingle
  };
}

exports["default"] = useSingle;
