// tdesign-vue-next/cjs/upload/themes/custom-file.js
/**
 * tdesign v1.11.1
 * (c) 2025 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var _defineProperty = require('@babel/runtime/helpers/defineProperty');
var Vue = require('vue');
var upload_hooks_useDrag = require('../hooks/useDrag.js');
var upload_constants = require('../constants.js');
var hooks_tnode = require('../../hooks/tnode.js');
require('../../_chunks/dep-733d16d5.js');
require('../../_chunks/dep-02147c0b.js');
require('../../utils/render-tnode.js');
require('../../_chunks/dep-ae77d744.js');
require('@babel/runtime/helpers/typeof');
require('../../_chunks/dep-acdd3481.js');
require('../../_chunks/dep-8af7b432.js');
require('../../_chunks/dep-fd23ce18.js');
require('../../_chunks/dep-750b6c90.js');
require('../../_chunks/dep-9a234926.js');
require('../../_chunks/dep-179c4586.js');
require('../../_chunks/dep-65ed94ad.js');
require('../../_chunks/dep-93deee44.js');
require('../../_chunks/dep-b74aef17.js');
require('../../_chunks/dep-53d908b3.js');
require('../../_chunks/dep-172ba234.js');
require('../../_chunks/dep-f3362596.js');
require('../../_chunks/dep-2930dcc3.js');
require('../../_chunks/dep-481d05e9.js');
require('../../_chunks/dep-ec56dcd9.js');
require('../../_chunks/dep-b402ceae.js');
require('../../_chunks/dep-901ff2b1.js');
require('../../_chunks/dep-b8cfc98c.js');
require('../../_chunks/dep-0d790559.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var _defineProperty__default = /*#__PURE__*/_interopDefaultLegacy(_defineProperty);

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty__default["default"](e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var CustomFile = Vue.defineComponent({
  name: "UploadCustomFile",
  props: _objectSpread(_objectSpread({}, upload_constants.commonProps), {}, {
    dragEvents: Object,
    draggable: Boolean,
    dragContent: Function,
    trigger: Function,
    triggerUpload: Function,
    childrenNode: [String, Function]
  }),
  setup: function setup(props, _ref) {
    var slots = _ref.slots;
    var _toRefs = Vue.toRefs(props),
      classPrefix = _toRefs.classPrefix,
      displayFiles = _toRefs.displayFiles,
      accept = _toRefs.accept;
    var drag = upload_hooks_useDrag["default"](props.dragEvents, accept);
    var dragActive = drag.dragActive;
    var renderContent = hooks_tnode.useContent();
    var renderDragContent = function renderDragContent() {
      var _props$childrenNode;
      var params = {
        dragActive: dragActive.value || false,
        files: displayFiles.value
      };
      return Vue.createVNode("div", {
        "class": "".concat(classPrefix.value, "-upload__dragger ").concat(classPrefix.value, "-upload__dragger-center"),
        "onDrop": drag.handleDrop,
        "onDragenter": drag.handleDragenter,
        "onDragover": drag.handleDragover,
        "onDragleave": drag.handleDragleave,
        "onClick": props.triggerUpload
      }, [Vue.createVNode("div", {
        "class": "".concat(classPrefix.value, "-upload__trigger")
      }, [renderContent("dragContent", "trigger", {
        params: params
      }) || ((_props$childrenNode = props.childrenNode) === null || _props$childrenNode === void 0 ? void 0 : _props$childrenNode.call(props, params))])]);
    };
    return function () {
      var _props$childrenNode2, _slots$default;
      return Vue.createVNode(Vue.Fragment, null, [props.draggable ? renderDragContent() : Vue.createVNode("div", {
        "class": "".concat(classPrefix.value, "-upload__trigger"),
        "onClick": props.triggerUpload
      }, [((_props$childrenNode2 = props.childrenNode) === null || _props$childrenNode2 === void 0 ? void 0 : _props$childrenNode2.call(props, {
        files: displayFiles.value
      })) || ((_slots$default = slots["default"]) === null || _slots$default === void 0 ? void 0 : _slots$default.call(slots))])]);
    };
  }
});

exports["default"] = CustomFile;
