// tdesign-vue-next/cjs/upload/themes/dragger-file.js
/**
 * tdesign v1.11.1
 * (c) 2025 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var _toConsumableArray = require('@babel/runtime/helpers/toConsumableArray');
var _defineProperty = require('@babel/runtime/helpers/defineProperty');
var Vue = require('vue');
var tdesignIconsVueNext = require('tdesign-icons-vue-next');
var utils = require('../../_chunks/dep-733d16d5.js');
var button_index = require('../../button/index.js');
var upload_constants = require('../constants.js');
var hooks_useCommonClassName = require('../../hooks/useCommonClassName.js');
var loading_index = require('../../loading/index.js');
var upload_hooks_useDrag = require('../hooks/useDrag.js');
var hooks_useGlobalIcon = require('../../hooks/useGlobalIcon.js');
var imageViewer_index = require('../../image-viewer/index.js');
require('@babel/runtime/helpers/typeof');
var hooks_tnode = require('../../hooks/tnode.js');
require('../../config-provider/useConfig.js');
require('../../hooks/useKeepAnimation.js');
require('@babel/runtime/helpers/slicedToArray');
var image_index = require('../../image/index.js');
require('../../_chunks/dep-02147c0b.js');
require('../../button/button.js');
require('../../button/props.js');
require('../../hooks/useRipple.js');
require('../../hooks/useConfig.js');
require('../../_chunks/dep-bc99882c.js');
require('../../_chunks/dep-dbac77fd.js');
require('../../_chunks/dep-c0c71352.js');
require('dayjs');
require('../../_chunks/dep-9e37eebf.js');
require('../../_chunks/dep-a01b02fa.js');
require('../../_chunks/dep-7b70fa2b.js');
require('../../_chunks/dep-901ff2b1.js');
require('../../_chunks/dep-f3362596.js');
require('../../_chunks/dep-8af7b432.js');
require('../../_chunks/dep-ae77d744.js');
require('../../_chunks/dep-8db04cd4.js');
require('../../_chunks/dep-0d790559.js');
require('../../_chunks/dep-750b6c90.js');
require('../../_chunks/dep-fd23ce18.js');
require('../../_chunks/dep-ec56dcd9.js');
require('../../_chunks/dep-b402ceae.js');
require('../../_chunks/dep-0d817d77.js');
require('../../_chunks/dep-bbc5435a.js');
require('../../_chunks/dep-10ffbad0.js');
require('../../_chunks/dep-89db45fd.js');
require('../../_chunks/dep-b75e3209.js');
require('../../_chunks/dep-55ae3cb5.js');
require('../../_chunks/dep-481d05e9.js');
require('../../_chunks/dep-b8cfc98c.js');
require('../../_chunks/dep-dc345794.js');
require('../../_chunks/dep-acdd3481.js');
require('../../_chunks/dep-dca6be9c.js');
require('../../hooks/useDisabled.js');
require('../../_chunks/dep-5c1c1647.js');
require('../../_chunks/dep-f6db9929.js');
require('../../loading/plugin.js');
require('../../_chunks/dep-ade5be71.js');
require('../../loading/icon/gradient.js');
require('../../_chunks/dep-623661fb.js');
require('@babel/runtime/helpers/objectWithoutProperties');
require('../../_chunks/dep-14f725de.js');
require('../../_chunks/dep-31fe118c.js');
require('../../_chunks/dep-db2cd3d2.js');
require('../../utils/dom.js');
require('../../utils/render-tnode.js');
require('../../_chunks/dep-9a234926.js');
require('../../_chunks/dep-179c4586.js');
require('../../_chunks/dep-65ed94ad.js');
require('../../_chunks/dep-93deee44.js');
require('../../_chunks/dep-b74aef17.js');
require('../../_chunks/dep-53d908b3.js');
require('../../_chunks/dep-172ba234.js');
require('../../_chunks/dep-2930dcc3.js');
require('../../loading/props.js');
require('../../hooks/useTeleport.js');
require('../../_chunks/dep-a246b8d5.js');
require('../../_chunks/dep-d9378865.js');
require('../../_chunks/dep-3a398159.js');
require('../../_chunks/dep-470ac849.js');
require('../../_chunks/dep-973f44c6.js');
require('../../_chunks/dep-b377c7d2.js');
require('../../_chunks/dep-029e312d.js');
require('../../utils/withInstall.js');
require('../../image-viewer/image-viewer.js');
require('../../hooks/useDefaultValue.js');
require('../../hooks/usePopupManager.js');
require('@babel/runtime/helpers/createClass');
require('@babel/runtime/helpers/classCallCheck');
require('../../hooks/useVModel.js');
require('../../image-viewer/base/ImageItem.js');
require('@babel/runtime/helpers/asyncToGenerator');
require('@babel/runtime/regenerator');
require('../../hooks/useImagePreviewUrl.js');
require('../../image-viewer/hooks.js');
require('../../_chunks/dep-23173a3f.js');
require('../../_chunks/dep-8ddb1659.js');
require('../../_chunks/dep-4396f8d3.js');
require('../../_chunks/dep-ff66bcee.js');
require('../../_chunks/dep-1c334ee9.js');
require('../../image-viewer/base/ImageModalIcon.js');
require('../../image-viewer/base/ImageViewerModal.js');
require('../../dialog/index.js');
require('../../dialog/dialog.js');
require('../../dialog/props.js');
require('../../dialog/hooks.js');
require('../../_chunks/dep-3a71bf71.js');
require('../../_chunks/dep-2e858700.js');
require('../../_chunks/dep-438edd07.js');
require('../../hooks/useDestroyOnClose.js');
require('../../_chunks/dep-06c4a440.js');
require('../../dialog/dialog-card.js');
require('../../dialog/dialog-card-props.js');
require('../../dialog/plugin.js');
require('../../image-viewer/props.js');
require('../../image-viewer/base/ImageViewerUtils.js');
require('../../tooltip/index.js');
require('../../tooltip/tooltip.js');
require('../../tooltip/props.js');
require('../../popup/props.js');
require('../../popup/index.js');
require('../../popup/popup.js');
require('@popperjs/core');
require('../../popup/container.js');
require('../../hooks/useResizeObserver.js');
require('../../tooltip/utils/index.js');
require('../../image-viewer/utils.js');
require('../../image-viewer/const.js');
require('../../image/image.js');
require('../../_chunks/dep-05614339.js');
require('../../image/props.js');
require('../../space/index.js');
require('../../space/space.js');
require('../../space/props.js');
require('../../hooks/slot.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var _toConsumableArray__default = /*#__PURE__*/_interopDefaultLegacy(_toConsumableArray);
var _defineProperty__default = /*#__PURE__*/_interopDefaultLegacy(_defineProperty);

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty__default["default"](e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var DraggerFile = Vue.defineComponent({
  name: "UploadDraggerFile",
  props: _objectSpread(_objectSpread({}, upload_constants.commonProps), {}, {
    trigger: Function,
    triggerUpload: Function,
    uploadFiles: Function,
    cancelUpload: Function,
    dragEvents: Object
  }),
  setup: function setup(props, _ref) {
    var slots = _ref.slots;
    var _toRefs = Vue.toRefs(props),
      displayFiles = _toRefs.displayFiles,
      disabled = _toRefs.disabled,
      accept = _toRefs.accept;
    var locale = Vue.computed(function () {
      return props.locale;
    });
    var renderTNodeJSX = hooks_tnode.useTNodeJSX();
    var _useCommonClassName = hooks_useCommonClassName["default"](),
      sizeClassNames = _useCommonClassName.sizeClassNames;
    var uploadPrefix = "".concat(props.classPrefix, "-upload");
    var drag = upload_hooks_useDrag["default"](props.dragEvents, accept);
    var dragActive = drag.dragActive;
    var draggerFileRef = Vue.ref();
    var classes = Vue.computed(function () {
      var _displayFiles$value$;
      return ["".concat(uploadPrefix, "__dragger"), _defineProperty__default["default"]({}, "".concat(uploadPrefix, "__dragger-center"), !displayFiles.value[0]), _defineProperty__default["default"]({}, "".concat(uploadPrefix, "__dragger-error"), ((_displayFiles$value$ = displayFiles.value[0]) === null || _displayFiles$value$ === void 0 ? void 0 : _displayFiles$value$.status) === "fail")];
    });
    var _useGlobalIcon = hooks_useGlobalIcon.useGlobalIcon({
        CheckCircleFilledIcon: tdesignIconsVueNext.CheckCircleFilledIcon,
        ErrorCircleFilledIcon: tdesignIconsVueNext.ErrorCircleFilledIcon
      }),
      CheckCircleFilledIcon = _useGlobalIcon.CheckCircleFilledIcon,
      ErrorCircleFilledIcon = _useGlobalIcon.ErrorCircleFilledIcon;
    var renderImage = function renderImage() {
      var _file$response;
      if (!props.displayFiles.length) return;
      var file = displayFiles.value[0];
      if (!file) return null;
      var url = (file === null || file === void 0 ? void 0 : file.url) || (file === null || file === void 0 || (_file$response = file.response) === null || _file$response === void 0 ? void 0 : _file$response.url);
      return Vue.createVNode("div", {
        "class": "".concat(uploadPrefix, "__dragger-img-wrap")
      }, [Vue.createVNode(imageViewer_index.ImageViewer, Vue.mergeProps({
        "images": [url],
        "trigger": function trigger(h2, _ref4) {
          var open = _ref4.open;
          return Vue.createVNode(image_index.Image, {
            "src": url || file.raw,
            "onClick": open,
            "error": "",
            "loading": ""
          }, null);
        }
      }, props.imageViewerProps), null)]);
    };
    var renderUploading = function renderUploading() {
      if (!props.displayFiles.length) return;
      var file = displayFiles.value[0];
      if (!file) return null;
      if ((file === null || file === void 0 ? void 0 : file.status) === "progress") {
        return Vue.createVNode("div", {
          "class": "".concat(uploadPrefix, "__single-progress")
        }, [Vue.createVNode(loading_index.Loading, null, null), props.showUploadProgress && Vue.createVNode("span", {
          "class": "".concat(uploadPrefix, "__single-percent")
        }, [file.percent, "%"])]);
      }
    };
    var renderMainPreview = function renderMainPreview() {
      var _locale$value;
      var file = displayFiles.value[0];
      var fileName = props.abridgeName ? utils.abridgeName.apply(void 0, [file.name].concat(_toConsumableArray__default["default"](props.abridgeName))) : file.name;
      var fileInfo = [Vue.createVNode("div", {
        "class": "".concat(uploadPrefix, "__dragger-text"),
        "key": "info"
      }, [Vue.createVNode("span", {
        "class": "".concat(uploadPrefix, "__single-name")
      }, [fileName]), file.status === "progress" && renderUploading(), file.status === "success" && Vue.createVNode(CheckCircleFilledIcon, null, null), file.status === "fail" && Vue.createVNode(ErrorCircleFilledIcon, null, null)]), Vue.createVNode("small", {
        "class": "".concat(sizeClassNames.small),
        "key": "size"
      }, [locale.value.file.fileSizeText, "\uFF1A", utils.getFileSizeText(file.size)]), Vue.createVNode("small", {
        "class": "".concat(sizeClassNames.small),
        "key": "time"
      }, [locale.value.file.fileOperationDateText, "\uFF1A", file.uploadTime || "-"])];
      return Vue.createVNode("div", {
        "class": "".concat(uploadPrefix, "__dragger-progress")
      }, [props.theme === "image" && renderImage(), Vue.createVNode("div", {
        "class": "".concat(uploadPrefix, "__dragger-progress-info")
      }, [renderTNodeJSX("fileListDisplay", {
        params: {
          files: props.displayFiles
        }
      }) || fileInfo, Vue.createVNode("div", {
        "class": "".concat(uploadPrefix, "__dragger-btns")
      }, [["progress", "waiting"].includes(file.status) && !disabled.value && Vue.createVNode(button_index.Button, {
        "theme": "primary",
        "variant": "text",
        "class": "".concat(uploadPrefix, "__dragger-progress-cancel"),
        "onClick": function onClick(e) {
          var _props$cancelUpload;
          return (_props$cancelUpload = props.cancelUpload) === null || _props$cancelUpload === void 0 ? void 0 : _props$cancelUpload.call(props, {
            e: e,
            file: props.toUploadFiles[0] || props.files[0]
          });
        }
      }, {
        "default": function _default() {
          return [(_locale$value = locale.value) === null || _locale$value === void 0 ? void 0 : _locale$value.cancelUploadText];
        }
      }), !props.autoUpload && file.status === "waiting" && Vue.createVNode(button_index.Button, {
        "theme": "primary",
        "variant": "text",
        "disabled": disabled.value,
        "onClick": function onClick() {
          var _props$uploadFiles;
          return (_props$uploadFiles = props.uploadFiles) === null || _props$uploadFiles === void 0 ? void 0 : _props$uploadFiles.call(props);
        },
        "class": "".concat(uploadPrefix, "__dragger-upload-btn")
      }, {
        "default": function _default() {
          return [locale.value.triggerUploadText.normal];
        }
      })]), ["fail", "success"].includes(file === null || file === void 0 ? void 0 : file.status) && !disabled.value && Vue.createVNode("div", {
        "class": "".concat(uploadPrefix, "__dragger-btns")
      }, [Vue.createVNode(button_index.Button, {
        "theme": "primary",
        "variant": "text",
        "disabled": disabled.value,
        "class": "".concat(uploadPrefix, "__dragger-progress-cancel"),
        "onClick": props.triggerUpload
      }, {
        "default": function _default() {
          return [locale.value.triggerUploadText.reupload];
        }
      }), Vue.createVNode(button_index.Button, {
        "theme": "danger",
        "variant": "text",
        "disabled": disabled.value,
        "class": "".concat(uploadPrefix, "__dragger-delete-btn"),
        "onClick": function onClick(e) {
          return props.onRemove({
            e: e,
            index: 0,
            file: file
          });
        }
      }, {
        "default": function _default() {
          return [locale.value.triggerUploadText["delete"]];
        }
      })])])]);
    };
    var renderDefaultDragElement = function renderDefaultDragElement() {
      var _locale$value$trigger;
      var unActiveElement = Vue.createVNode("div", null, [Vue.createVNode("span", {
        "class": "".concat(uploadPrefix, "--highlight")
      }, [(_locale$value$trigger = locale.value.triggerUploadText) === null || _locale$value$trigger === void 0 ? void 0 : _locale$value$trigger.normal]), Vue.createVNode("span", null, ["\xA0\xA0/\xA0\xA0", locale.value.dragger.draggingText])]);
      var activeElement = Vue.createVNode("div", null, [locale.value.dragger.dragDropText]);
      return dragActive.value ? activeElement : unActiveElement;
    };
    var getContent = function getContent() {
      var _slots$default;
      var file = displayFiles.value[0];
      if (file && (["progress", "success", "fail", "waiting"].includes(file.status) || !file.status)) {
        return renderMainPreview();
      }
      return Vue.createVNode("div", {
        "class": "".concat(uploadPrefix, "__trigger"),
        "onClick": props.triggerUpload
      }, [((_slots$default = slots["default"]) === null || _slots$default === void 0 ? void 0 : _slots$default.call(slots)) || renderDefaultDragElement()]);
    };
    return function () {
      var _props$trigger;
      return Vue.createVNode("div", {
        "ref": draggerFileRef,
        "class": classes.value,
        "onDrop": drag.handleDrop,
        "onDragenter": drag.handleDragenter,
        "onDragover": drag.handleDragover,
        "onDragleave": drag.handleDragleave
      }, [((_props$trigger = props.trigger) === null || _props$trigger === void 0 ? void 0 : _props$trigger.call(props, Vue.h, {
        files: displayFiles.value,
        dragActive: dragActive.value
      })) || getContent()]);
    };
  }
});

exports["default"] = DraggerFile;
