// tdesign-vue-next/cjs/hooks/useConfig.js
/**
 * tdesign v1.11.1
 * (c) 2025 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var configProvider_useConfig = require('../config-provider/useConfig.js');
require('../_chunks/dep-bc99882c.js');
require('../_chunks/dep-dbac77fd.js');
require('@babel/runtime/helpers/typeof');
require('../_chunks/dep-c0c71352.js');
require('dayjs');
require('../_chunks/dep-9e37eebf.js');
require('../_chunks/dep-a01b02fa.js');
require('../_chunks/dep-7b70fa2b.js');
require('../_chunks/dep-901ff2b1.js');
require('../_chunks/dep-f3362596.js');
require('../_chunks/dep-8af7b432.js');
require('../_chunks/dep-ae77d744.js');
require('../_chunks/dep-8db04cd4.js');
require('../_chunks/dep-0d790559.js');
require('../_chunks/dep-750b6c90.js');
require('../_chunks/dep-fd23ce18.js');
require('../_chunks/dep-ec56dcd9.js');
require('../_chunks/dep-b402ceae.js');
require('../_chunks/dep-0d817d77.js');
require('../_chunks/dep-bbc5435a.js');
require('../_chunks/dep-10ffbad0.js');
require('../_chunks/dep-89db45fd.js');
require('../_chunks/dep-b75e3209.js');
require('../_chunks/dep-55ae3cb5.js');
require('../_chunks/dep-481d05e9.js');
require('../_chunks/dep-b8cfc98c.js');
require('../_chunks/dep-dc345794.js');
require('../_chunks/dep-acdd3481.js');

function usePrefixClass(componentName) {
  var _useConfig = configProvider_useConfig.useConfig("classPrefix"),
    classPrefix = _useConfig.classPrefix;
  return Vue.computed(function () {
    return componentName ? "".concat(classPrefix.value, "-").concat(componentName) : classPrefix.value;
  });
}
function useCommonClassName() {
  var _useConfig2 = configProvider_useConfig.useConfig("classPrefix"),
    classPrefix = _useConfig2.classPrefix;
  return {
    SIZE: Vue.computed(function () {
      return {
        small: "".concat(classPrefix.value, "-size-s"),
        medium: "".concat(classPrefix.value, "-size-m"),
        large: "".concat(classPrefix.value, "-size-l"),
        "default": "",
        xs: "".concat(classPrefix.value, "-size-xs"),
        xl: "".concat(classPrefix.value, "-size-xl"),
        block: "".concat(classPrefix.value, "-size-full-width")
      };
    }),
    STATUS: Vue.computed(function () {
      return {
        loading: "".concat(classPrefix.value, "-is-loading"),
        loadMore: "".concat(classPrefix.value, "-is-load-more"),
        disabled: "".concat(classPrefix.value, "-is-disabled"),
        focused: "".concat(classPrefix.value, "-is-focused"),
        success: "".concat(classPrefix.value, "-is-success"),
        error: "".concat(classPrefix.value, "-is-error"),
        warning: "".concat(classPrefix.value, "-is-warning"),
        selected: "".concat(classPrefix.value, "-is-selected"),
        active: "".concat(classPrefix.value, "-is-active"),
        checked: "".concat(classPrefix.value, "-is-checked"),
        current: "".concat(classPrefix.value, "-is-current"),
        hidden: "".concat(classPrefix.value, "-is-hidden"),
        visible: "".concat(classPrefix.value, "-is-visible"),
        expanded: "".concat(classPrefix.value, "-is-expanded"),
        indeterminate: "".concat(classPrefix.value, "-is-indeterminate")
      };
    })
  };
}

exports.useConfig = configProvider_useConfig.useConfig;
exports.useCommonClassName = useCommonClassName;
exports.usePrefixClass = usePrefixClass;
