// tdesign-vue-next/cjs/color-picker/panel/index.js
/**
 * tdesign v1.11.1
 * (c) 2025 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var _toConsumableArray = require('@babel/runtime/helpers/toConsumableArray');
var _slicedToArray = require('@babel/runtime/helpers/slicedToArray');
var _defineProperty = require('@babel/runtime/helpers/defineProperty');
var Vue = require('vue');
var hooks_useConfig = require('../../hooks/useConfig.js');
var colorPicker_props = require('../props.js');
var constants = require('../../_chunks/dep-acf307a4.js');
var colorPicker_panel_header = require('./header.js');
var colorPicker_panel_linearGradient = require('./linear-gradient.js');
var colorPicker_panel_saturation = require('./saturation.js');
var colorPicker_panel_hue = require('./hue.js');
var colorPicker_panel_alpha = require('./alpha.js');
var colorPicker_panel_format_index = require('./format/index.js');
var colorPicker_panel_swatches = require('./swatches.js');
var color = require('../../_chunks/dep-43b5d648.js');
require('@babel/runtime/helpers/classCallCheck');
require('@babel/runtime/helpers/createClass');
var colorPicker_hooks_index = require('../hooks/index.js');
var hooks_useVModel = require('../../hooks/useVModel.js');
var hooks_useDefaultValue = require('../../hooks/useDefaultValue.js');
var configProvider_useConfig = require('../../config-provider/useConfig.js');
var cloneDeep = require('../../_chunks/dep-b75e3209.js');
require('../../_chunks/dep-bc99882c.js');
require('../../_chunks/dep-dbac77fd.js');
require('@babel/runtime/helpers/typeof');
require('../../_chunks/dep-c0c71352.js');
require('dayjs');
require('../../_chunks/dep-9e37eebf.js');
require('../../_chunks/dep-a01b02fa.js');
require('../../_chunks/dep-7b70fa2b.js');
require('../../_chunks/dep-901ff2b1.js');
require('../../_chunks/dep-f3362596.js');
require('../../_chunks/dep-8af7b432.js');
require('../../_chunks/dep-ae77d744.js');
require('../../_chunks/dep-8db04cd4.js');
require('../../_chunks/dep-0d790559.js');
require('../../_chunks/dep-750b6c90.js');
require('../../_chunks/dep-fd23ce18.js');
require('../../_chunks/dep-ec56dcd9.js');
require('../../_chunks/dep-b402ceae.js');
require('../../_chunks/dep-0d817d77.js');
require('../../_chunks/dep-bbc5435a.js');
require('../../_chunks/dep-10ffbad0.js');
require('../../_chunks/dep-89db45fd.js');
require('../../_chunks/dep-55ae3cb5.js');
require('../../_chunks/dep-481d05e9.js');
require('../../_chunks/dep-b8cfc98c.js');
require('../../_chunks/dep-dc345794.js');
require('../../_chunks/dep-acdd3481.js');
require('../../radio/index.js');
require('../../radio/radio.js');
require('../../radio/props.js');
require('../../radio/consts/index.js');
require('../../hooks/useDisabled.js');
require('../../_chunks/dep-5c1c1647.js');
require('../../hooks/useReadonly.js');
require('../../hooks/tnode.js');
require('../../utils/render-tnode.js');
require('../../_chunks/dep-9a234926.js');
require('../../_chunks/dep-179c4586.js');
require('../../_chunks/dep-65ed94ad.js');
require('../../_chunks/dep-93deee44.js');
require('../../_chunks/dep-b74aef17.js');
require('../../_chunks/dep-53d908b3.js');
require('../../_chunks/dep-172ba234.js');
require('../../_chunks/dep-2930dcc3.js');
require('../../_chunks/dep-623661fb.js');
require('@babel/runtime/helpers/objectWithoutProperties');
require('../../_chunks/dep-14f725de.js');
require('../../_chunks/dep-31fe118c.js');
require('../../_chunks/dep-db2cd3d2.js');
require('../../_chunks/dep-3a71bf71.js');
require('../../_chunks/dep-973f44c6.js');
require('../../_chunks/dep-2e858700.js');
require('../../_chunks/dep-438edd07.js');
require('../../radio/group.js');
require('@babel/runtime/helpers/asyncToGenerator');
require('@babel/runtime/regenerator');
require('../../radio/radio-group-props.js');
require('../../radio/radio-button.js');
require('../../radio/hooks/useKeyboard.js');
require('../../utils/dom.js');
require('../../_chunks/dep-d2dcffb5.js');
require('../../watermark/hooks/index.js');
require('../../hooks/useResizeObserver.js');
require('../../_chunks/dep-4396f8d3.js');
require('../../_chunks/dep-ff66bcee.js');
require('../../_chunks/dep-1c334ee9.js');
require('../../_chunks/dep-3f6b2b9c.js');
require('../../utils/withInstall.js');
require('../../hooks/useKeepAnimation.js');
require('../../input-number/index.js');
require('../../input-number/input-number.js');
require('tdesign-icons-vue-next');
require('../../button/index.js');
require('../../button/button.js');
require('../../loading/index.js');
require('../../_chunks/dep-f6db9929.js');
require('../../loading/plugin.js');
require('../../_chunks/dep-ade5be71.js');
require('../../loading/icon/gradient.js');
require('../../_chunks/dep-dca6be9c.js');
require('../../loading/props.js');
require('../../hooks/useTeleport.js');
require('../../_chunks/dep-a246b8d5.js');
require('../../_chunks/dep-d9378865.js');
require('../../_chunks/dep-3a398159.js');
require('../../_chunks/dep-470ac849.js');
require('../../_chunks/dep-b377c7d2.js');
require('../../_chunks/dep-029e312d.js');
require('../../button/props.js');
require('../../hooks/useRipple.js');
require('../../input/index.js');
require('../../input/input.js');
require('../../input/props.js');
require('../../hooks/useGlobalIcon.js');
require('../../input/useInput.js');
require('../../form/const.js');
require('../../input/useLengthLimit.js');
require('../../_chunks/dep-02147c0b.js');
require('../../input/useInputEventHandler.js');
require('../../input/useInputWidth.js');
require('../../input/input-group.js');
require('../../input/input-group-props.js');
require('../../input-number/props.js');
require('../../input-number/hooks/useInputNumber.js');
require('../../hooks/useCommonClassName.js');
require('../../_chunks/dep-23173a3f.js');
require('../../_chunks/dep-8ddb1659.js');
require('./base-props.js');
require('../../_chunks/dep-eeb451cb.js');
require('./slider.js');
require('../../select/index.js');
require('../../select/select.js');
require('../../common-components/fake-arrow.js');
require('../../select-input/index.js');
require('../../select-input/select-input.js');
require('../../popup/index.js');
require('../../popup/popup.js');
require('@popperjs/core');
require('../../popup/container.js');
require('../../popup/props.js');
require('../../select-input/props.js');
require('../../select-input/useSingle.js');
require('../../_chunks/dep-eee7af37.js');
require('../../_chunks/dep-b694cee7.js');
require('../../select-input/useMultiple.js');
require('../../tag-input/index.js');
require('../../tag-input/tag-input.js');
require('../../tag-input/props.js');
require('../../tag-input/hooks/useDragSorter.js');
require('../../tag-input/hooks/useHover.js');
require('../../tag-input/hooks/useTagScroll.js');
require('../../tag-input/hooks/useTagList.js');
require('../../tag/index.js');
require('../../tag/tag.js');
require('tinycolor2');
require('../../tag/props.js');
require('../../tag/check-tag.js');
require('../../tag/check-tag-props.js');
require('../../tag/check-tag-group.js');
require('../../tag/check-tag-group-props.js');
require('../../select-input/useOverlayInnerStyle.js');
require('../../select/select-panel.js');
require('../../select/option.js');
require('../../select/option-props.js');
require('../../checkbox/index.js');
require('../../checkbox/checkbox.js');
require('../../checkbox/props.js');
require('../../checkbox/consts/index.js');
require('../../checkbox/hooks/useCheckboxLazyLoad.js');
require('../../_chunks/dep-05614339.js');
require('../../checkbox/hooks/useKeyboardEvent.js');
require('../../checkbox/group.js');
require('../../checkbox/checkbox-group-props.js');
require('../../hooks/slot.js');
require('../../_chunks/dep-a0482c90.js');
require('../../_chunks/dep-0489844a.js');
require('../../select/utils/index.js');
require('../../select/consts/index.js');
require('../../select/option-group.js');
require('../../select/option-group-props.js');
require('../../select/props.js');
require('../../select/hooks/usePanelVirtualScroll.js');
require('../../hooks/useVirtualScroll.js');
require('../../select/hooks/useKeyboardControl.js');
require('../../select/hooks/useSelectOptions.js');
require('./format/inputs.js');
require('./format/config.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var _toConsumableArray__default = /*#__PURE__*/_interopDefaultLegacy(_toConsumableArray);
var _slicedToArray__default = /*#__PURE__*/_interopDefaultLegacy(_slicedToArray);
var _defineProperty__default = /*#__PURE__*/_interopDefaultLegacy(_defineProperty);

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty__default["default"](e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var ColorPanel = Vue.defineComponent({
  name: "ColorPanel",
  props: _objectSpread(_objectSpread({}, colorPicker_props["default"]), {}, {
    togglePopup: {
      type: Function
    }
  }),
  setup: function setup(props2) {
    var _props2$colorModes, _innerValue$value, _props2$colorModes2;
    var baseClassName = colorPicker_hooks_index.useBaseClassName();
    var _useCommonClassName = hooks_useConfig.useCommonClassName(),
      STATUS = _useCommonClassName.STATUS;
    var _useConfig = configProvider_useConfig.useConfig("colorPicker"),
      t = _useConfig.t,
      globalConfig = _useConfig.globalConfig;
    var statusClassNames = STATUS.value;
    var _toRefs = Vue.toRefs(props2),
      inputValue = _toRefs.value,
      modelValue = _toRefs.modelValue,
      recentColors = _toRefs.recentColors;
    var _useVModel = hooks_useVModel["default"](inputValue, modelValue, props2.defaultValue, props2.onChange),
      _useVModel2 = _slicedToArray__default["default"](_useVModel, 2),
      innerValue = _useVModel2[0],
      setInnerValue = _useVModel2[1];
    var defaultEmptyColor = Vue.computed(function () {
      return isGradient.value ? constants.DEFAULT_LINEAR_GRADIENT : constants.DEFAULT_COLOR;
    });
    var mode = Vue.ref(((_props2$colorModes = props2.colorModes) === null || _props2$colorModes === void 0 ? void 0 : _props2$colorModes.length) !== 1 && (_innerValue$value = innerValue.value) !== null && _innerValue$value !== void 0 && _innerValue$value.includes("linear-gradient") ? "linear-gradient" : ((_props2$colorModes2 = props2.colorModes) === null || _props2$colorModes2 === void 0 ? void 0 : _props2$colorModes2.length) === 1 ? props2.colorModes[0] : "monochrome");
    var isGradient = Vue.computed(function () {
      return mode.value === "linear-gradient";
    });
    var color$1 = Vue.ref(new color.Color(innerValue.value || defaultEmptyColor.value));
    var updateColor = function updateColor() {
      return color$1.value.update(innerValue.value || defaultEmptyColor.value);
    };
    var formatModel = Vue.ref(color$1.value.isGradient ? "CSS" : "RGB");
    var _useDefaultValue = hooks_useDefaultValue["default"](recentColors, props2.defaultRecentColors, props2.onRecentColorsChange, "recentColors"),
      _useDefaultValue2 = _slicedToArray__default["default"](_useDefaultValue, 2),
      recentlyUsedColors = _useDefaultValue2[0],
      setRecentlyUsedColors = _useDefaultValue2[1];
    var formatValue = function formatValue() {
      if (mode.value === "linear-gradient") {
        return color$1.value.linearGradient;
      }
      var colorMap = color$1.value.getFormatsColorMap();
      return Object.keys(colorMap).includes(props2.format) ? colorMap[props2.format] : color$1.value.css;
    };
    var addRecentlyUsedColor = function addRecentlyUsedColor() {
      if (recentlyUsedColors.value === null || recentlyUsedColors.value === false) {
        return;
      }
      var colors = cloneDeep.cloneDeep(recentlyUsedColors.value) || [];
      var currentColor = color$1.value.isGradient ? color$1.value.linearGradient : color$1.value.rgba;
      var index = colors.indexOf(currentColor);
      if (index > -1) {
        colors.splice(index, 1);
      }
      colors.unshift(currentColor);
      if (colors.length > constants.TD_COLOR_USED_COLORS_MAX_SIZE) {
        colors.length = constants.TD_COLOR_USED_COLORS_MAX_SIZE;
      }
      handleRecentlyUsedColorsChange(colors);
    };
    var handleRecentlyUsedColorsChange = function handleRecentlyUsedColorsChange(colors) {
      setRecentlyUsedColors(colors);
    };
    var emitColorChange = function emitColorChange(trigger) {
      setInnerValue(formatValue(), {
        color: color.getColorObject(color$1.value),
        trigger: trigger || "palette-saturation-brightness"
      });
    };
    Vue.watch(function () {
      return [props2.defaultValue, props2.enableAlpha];
    }, updateColor);
    Vue.watch(function () {
      return innerValue.value;
    }, function (newColor) {
      if (newColor !== formatValue()) {
        updateColor();
        mode.value = color$1.value.isGradient ? "linear-gradient" : "monochrome";
      }
    });
    var handleModeChange = function handleModeChange(value) {
      mode.value = value;
      if (value === "linear-gradient") {
        color$1.value.update(color$1.value.gradientColors.length > 0 ? color$1.value.linearGradient : constants.DEFAULT_LINEAR_GRADIENT);
      } else {
        color$1.value.update(color$1.value.rgba);
      }
      emitColorChange();
    };
    var handleFormatModeChange = function handleFormatModeChange(format) {
      return formatModel.value = format;
    };
    var handleSatAndValueChange = function handleSatAndValueChange(_ref) {
      var saturation = _ref.saturation,
        value = _ref.value;
      var _color$value = color$1.value,
        sat = _color$value.saturation,
        val = _color$value.value;
      var changeTrigger = "palette-saturation-brightness";
      if (value !== val && saturation !== sat) {
        color$1.value.saturation = saturation;
        color$1.value.value = value;
        changeTrigger = "palette-saturation-brightness";
      } else if (saturation !== sat) {
        color$1.value.saturation = saturation;
        changeTrigger = "palette-saturation";
      } else if (value !== val) {
        color$1.value.value = value;
        changeTrigger = "palette-brightness";
      } else {
        return;
      }
      color$1.value.update(color$1.value.rgba);
      emitColorChange(changeTrigger);
    };
    var handleHueChange = function handleHueChange(hue) {
      var _props2$onPaletteBarC;
      color$1.value.hue = hue;
      emitColorChange("palette-hue-bar");
      (_props2$onPaletteBarC = props2.onPaletteBarChange) === null || _props2$onPaletteBarC === void 0 || _props2$onPaletteBarC.call(props2, {
        color: color.getColorObject(color$1.value)
      });
    };
    var handleAlphaChange = function handleAlphaChange(alpha) {
      color$1.value.alpha = alpha;
      emitColorChange("palette-alpha-bar");
    };
    var handleInputChange = function handleInputChange(input, alpha) {
      color$1.value.update(input);
      color$1.value.alpha = alpha;
      emitColorChange("input");
    };
    var handleGradientChange = function handleGradientChange(_ref2) {
      var key = _ref2.key,
        payload = _ref2.payload;
      var trigger = "palette-saturation-brightness";
      switch (key) {
        case "degree":
          color$1.value.gradientDegree = payload;
          trigger = "input";
          break;
        case "selectedId":
          color$1.value.gradientSelectedId = payload;
          break;
        case "colors":
          color$1.value.gradientColors = payload;
          break;
      }
      emitColorChange(trigger);
    };
    var handleSetColor = function handleSetColor(type, value) {
      var isGradientValue = color.Color.isGradientColor(value);
      if (isGradientValue) {
        if (props2.colorModes.includes("linear-gradient")) {
          mode.value = "linear-gradient";
          color$1.value.update(value);
          color$1.value.updateCurrentGradientColor();
        } else {
          console.warn("\u8BE5\u6A21\u5F0F\u4E0D\u652F\u6301\u6E10\u53D8\u8272");
        }
      } else if (mode.value === "linear-gradient") {
        color$1.value.updateStates(value);
        color$1.value.updateCurrentGradientColor();
      } else {
        color$1.value.update(value);
      }
      emitColorChange();
    };
    return {
      baseClassName: baseClassName,
      statusClassNames: statusClassNames,
      t: t,
      globalConfig: globalConfig,
      color: color$1,
      mode: mode,
      formatModel: formatModel,
      recentlyUsedColors: recentlyUsedColors,
      isGradient: isGradient,
      addRecentlyUsedColor: addRecentlyUsedColor,
      handleModeChange: handleModeChange,
      handleSatAndValueChange: handleSatAndValueChange,
      handleHueChange: handleHueChange,
      handleAlphaChange: handleAlphaChange,
      handleGradientChange: handleGradientChange,
      handleSetColor: handleSetColor,
      handleFormatModeChange: handleFormatModeChange,
      handleInputChange: handleInputChange,
      handleRecentlyUsedColorsChange: handleRecentlyUsedColorsChange
    };
  },
  render: function render() {
    var _systemColors,
      _this = this;
    var t = this.t,
      baseClassName = this.baseClassName,
      statusClassNames = this.statusClassNames,
      globalConfig = this.globalConfig,
      swatchColors = this.swatchColors,
      showPrimaryColorPreview = this.showPrimaryColorPreview,
      isGradient = this.isGradient;
    var baseProps = {
      color: this.color,
      disabled: this.disabled
    };
    var showUsedColors = this.recentlyUsedColors !== null && this.recentlyUsedColors !== false;
    var systemColors = swatchColors;
    if (systemColors === void 0) {
      systemColors = _toConsumableArray__default["default"](constants.DEFAULT_SYSTEM_SWATCH_COLORS);
    }
    var showSystemColors = ((_systemColors = systemColors) === null || _systemColors === void 0 ? void 0 : _systemColors.length) > 0;
    var renderSwatches = function renderSwatches() {
      if (!showSystemColors && !showUsedColors) {
        return null;
      }
      return Vue.createVNode(Vue.Fragment, null, [Vue.createVNode("div", {
        "class": "".concat(baseClassName, "__swatches-wrap")
      }, [showUsedColors ? Vue.createVNode(colorPicker_panel_swatches["default"], Vue.mergeProps(baseProps, {
        "title": t(globalConfig.recentColorTitle),
        "editable": true,
        "colors": _this.recentlyUsedColors,
        "handleAddColor": _this.addRecentlyUsedColor,
        "onSetColor": function onSetColor(color) {
          return _this.handleSetColor("used", color);
        },
        "onChange": _this.handleRecentlyUsedColorsChange
      }), null) : null, showSystemColors ? Vue.createVNode(colorPicker_panel_swatches["default"], Vue.mergeProps(baseProps, {
        "title": t(globalConfig.swatchColorTitle),
        "colors": systemColors,
        "onSetColor": function onSetColor(color) {
          return _this.handleSetColor("system", color);
        }
      }), null) : null])]);
    };
    return Vue.createVNode("div", {
      "class": ["".concat(baseClassName, "__panel"), this.disabled ? statusClassNames.disabled : false]
    }, [Vue.createVNode(colorPicker_panel_header["default"], Vue.mergeProps(this.$props, {
      "mode": this.mode,
      "onModeChange": this.handleModeChange
    }), null), Vue.createVNode("div", {
      "class": ["".concat(baseClassName, "__body")]
    }, [isGradient ? Vue.createVNode(colorPicker_panel_linearGradient["default"], Vue.mergeProps(baseProps, {
      "onChange": this.handleGradientChange,
      "enableMultipleGradient": this.enableMultipleGradient
    }), null) : null, Vue.createVNode(colorPicker_panel_saturation["default"], Vue.mergeProps(baseProps, {
      "onChange": this.handleSatAndValueChange
    }), null), Vue.createVNode("div", {
      "class": ["".concat(baseClassName, "__sliders-wrapper")]
    }, [Vue.createVNode("div", {
      "class": ["".concat(baseClassName, "__sliders")]
    }, [Vue.createVNode(colorPicker_panel_hue["default"], Vue.mergeProps(baseProps, {
      "onChange": this.handleHueChange
    }), null), this.enableAlpha ? Vue.createVNode(colorPicker_panel_alpha["default"], Vue.mergeProps(baseProps, {
      "onChange": this.handleAlphaChange
    }), null) : null]), showPrimaryColorPreview ? Vue.createVNode("div", {
      "class": ["".concat(baseClassName, "__sliders-preview"), "".concat(baseClassName, "--bg-alpha")]
    }, [Vue.createVNode("span", {
      "class": "".concat(baseClassName, "__sliders-preview-inner"),
      "style": {
        background: isGradient ? this.color.linearGradient : this.color.rgba
      }
    }, null)]) : null]), Vue.createVNode(colorPicker_panel_format_index["default"], Vue.mergeProps(this.$props, {
      "color": this.color,
      "format": this.formatModel,
      "onModeChange": this.handleFormatModeChange,
      "onInputChange": this.handleInputChange
    }), null), renderSwatches()])]);
  }
});

exports["default"] = ColorPanel;
