// tdesign-vue-next/es/color-picker/panel/alpha.mjs
/**
 * tdesign v1.11.1
 * (c) 2025 tdesign
 * @license MIT
 */

import { defineComponent, computed, createVNode } from 'vue';
import ColorSlider from './slider.mjs';
import { useBaseClassName } from '../hooks/index.mjs';
import '../../_chunks/dep-96b403e3.mjs';
import '../../_chunks/dep-7c8701fc.mjs';
import '../../_chunks/dep-6760162b.mjs';
import '../utils/index.mjs';
import '../../_chunks/dep-a3e567cf.mjs';
import '../../_chunks/dep-6ac84404.mjs';
import '../../_chunks/dep-cf893fd5.mjs';
import '../../_chunks/dep-68c35191.mjs';
import 'lodash-es';
import './base-props.mjs';
import '../../hooks/useConfig.mjs';
import '../../config-provider/useConfig.mjs';
import '../../config-provider/context.mjs';
import '../../_chunks/dep-2895d202.mjs';
import '../../_chunks/dep-23a6b274.mjs';
import '../../_chunks/dep-f9079916.mjs';
import '../../_chunks/dep-c3b9ffc0.mjs';
import '../../config-provider/type.mjs';

var AlphaSlider = defineComponent({
  name: "AlphaSlider",
  inheritAttrs: false,
  props: {
    color: {
      type: Object
    },
    disabled: {
      type: Boolean,
      "default": false
    },
    onChange: {
      type: Function,
      "default": function _default() {
        return function () {};
      }
    }
  },
  setup: function setup(props) {
    var baseClassName = useBaseClassName();
    var handleChange = function handleChange(v, isDragEnd) {
      props.onChange(v / 100, isDragEnd);
    };
    var railStyle = computed(function () {
      return {
        background: "linear-gradient(to right, rgba(0, 0, 0, 0), ".concat(props.color.rgb, ")")
      };
    });
    return {
      baseClassName: baseClassName,
      railStyle: railStyle,
      handleChange: handleChange
    };
  },
  render: function render() {
    var baseClassName = this.baseClassName;
    return createVNode(ColorSlider, {
      "class": ["".concat(baseClassName, "__alpha"), "".concat(baseClassName, "--bg-alpha")],
      "color": this.color,
      "value": this.color.alpha * 100,
      "onChange": this.handleChange,
      "rail-style": this.railStyle,
      "max-value": 100,
      "disabled": this.disabled,
      "type": "alpha"
    }, null);
  }
});

export { AlphaSlider as default };
