// tdesign-vue-next/cjs/config-provider/useConfig.js
/**
 * tdesign v1.11.1
 * (c) 2025 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var configProvider_context = require('../_chunks/dep-bc99882c.js');
var cloneDeep = require('../_chunks/dep-b75e3209.js');
var isString = require('../_chunks/dep-acdd3481.js');
var isFunction = require('../_chunks/dep-f3362596.js');
require('../_chunks/dep-dbac77fd.js');
require('@babel/runtime/helpers/typeof');
require('../_chunks/dep-c0c71352.js');
require('dayjs');
require('../_chunks/dep-9e37eebf.js');
require('../_chunks/dep-a01b02fa.js');
require('../_chunks/dep-7b70fa2b.js');
require('../_chunks/dep-901ff2b1.js');
require('../_chunks/dep-8af7b432.js');
require('../_chunks/dep-ae77d744.js');
require('../_chunks/dep-8db04cd4.js');
require('../_chunks/dep-0d790559.js');
require('../_chunks/dep-750b6c90.js');
require('../_chunks/dep-fd23ce18.js');
require('../_chunks/dep-ec56dcd9.js');
require('../_chunks/dep-b402ceae.js');
require('../_chunks/dep-0d817d77.js');
require('../_chunks/dep-bbc5435a.js');
require('../_chunks/dep-10ffbad0.js');
require('../_chunks/dep-89db45fd.js');
require('../_chunks/dep-55ae3cb5.js');
require('../_chunks/dep-481d05e9.js');
require('../_chunks/dep-b8cfc98c.js');
require('../_chunks/dep-dc345794.js');

var globalConfigCopy = Vue.ref();
function useConfig() {
  var componentName = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : void 0;
  var componentLocale = arguments.length > 1 ? arguments[1] : undefined;
  var injectGlobalConfig = Vue.getCurrentInstance() ? Vue.inject(configProvider_context.configProviderInjectKey, null) : globalConfigCopy;
  var mergedGlobalConfig = Vue.computed(function () {
    return (injectGlobalConfig === null || injectGlobalConfig === void 0 ? void 0 : injectGlobalConfig.value) || configProvider_context.defaultGlobalConfig;
  });
  var globalConfig = Vue.computed(function () {
    return Object.assign({}, mergedGlobalConfig.value[componentName], componentLocale);
  });
  var classPrefix = Vue.computed(function () {
    return mergedGlobalConfig.value.classPrefix;
  });
  var t = function t(pattern) {
    for (var _len = arguments.length, args = new Array(_len > 1 ? _len - 1 : 0), _key = 1; _key < _len; _key++) {
      args[_key - 1] = arguments[_key];
    }
    var data = args[0];
    if (isString.isString(pattern)) {
      if (!data) return pattern;
      var regular = /\{\s*([\w-]+)\s*\}/g;
      var translated = pattern.replace(regular, function (match, key) {
        if (data) {
          return String(data[key]);
        }
        return "";
      });
      return translated;
    }
    if (isFunction.isFunction(pattern)) {
      if (!args.length) return pattern(Vue.h);
      return pattern.apply(void 0, args);
    }
    return "";
  };
  return {
    t: t,
    global: globalConfig,
    globalConfig: globalConfig,
    classPrefix: classPrefix
  };
}
var provideConfig = function provideConfig(props) {
  var defaultData = cloneDeep.cloneDeep(configProvider_context.defaultGlobalConfig);
  var mergedGlobalConfig = Vue.computed(function () {
    return Object.assign({}, configProvider_context.mergeWith(defaultData, props.globalConfig));
  });
  Vue.provide(configProvider_context.configProviderInjectKey, mergedGlobalConfig);
  if (!globalConfigCopy.value) {
    globalConfigCopy.value = mergedGlobalConfig.value;
  }
  return mergedGlobalConfig;
};

exports.provideConfig = provideConfig;
exports.useConfig = useConfig;
