// tdesign-vue-next/es/date-picker/panel/ExtraContent.mjs
/**
 * tdesign v1.11.1
 * (c) 2025 tdesign
 * @license MIT
 */

import { defineComponent, computed, createVNode } from 'vue';
import TDateFooter from '../base/Footer.mjs';
import '../../_chunks/dep-96b403e3.mjs';
import '../../_chunks/dep-7c8701fc.mjs';
import '../../hooks/useConfig.mjs';
import '../../config-provider/useConfig.mjs';
import 'lodash-es';
import '../../config-provider/context.mjs';
import '../../_chunks/dep-2895d202.mjs';
import '../../_chunks/dep-23a6b274.mjs';
import '../../_chunks/dep-f9079916.mjs';
import '../../_chunks/dep-c3b9ffc0.mjs';
import '../../config-provider/type.mjs';
import '../../button/index.mjs';
import '../../button/button.mjs';
import '../../_chunks/dep-87669a0e.mjs';
import '../../_chunks/dep-6ac84404.mjs';
import '../../loading/index.mjs';
import '../../loading/directive.mjs';
import '../../loading/plugin.mjs';
import '../../_chunks/dep-433c4f57.mjs';
import '../../_chunks/dep-a3e567cf.mjs';
import '../../loading/icon/gradient.mjs';
import '../../_chunks/dep-4ad0839f.mjs';
import '../../_chunks/dep-5211a143.mjs';
import '../../_chunks/dep-e155a416.mjs';
import '../../utils/dom.mjs';
import '../../utils/render-tnode.mjs';
import '../../loading/props.mjs';
import '../../hooks/useTeleport.mjs';
import '../../utils/withInstall.mjs';
import './style/css.mjs';
import '../../loading/type.mjs';
import '../../button/props.mjs';
import '../../hooks/useRipple.mjs';
import '../../hooks/useKeepAnimation.mjs';
import '../../hooks/tnode.mjs';
import '../../hooks/useDisabled.mjs';
import '../../button/type.mjs';

var TExtraContent = defineComponent({
  name: "TExtraContent",
  props: {
    presets: Object,
    enableTimePicker: Boolean,
    presetsPlacement: String,
    needConfirm: {
      type: Boolean,
      "default": true
    },
    onPresetClick: Function,
    onConfirmClick: Function,
    selectedValue: [String, Number, Array, Date]
  },
  setup: function setup(props) {
    var showPanelFooter = computed(function () {
      return props.enableTimePicker && props.needConfirm || props.presets;
    });
    return function () {
      return showPanelFooter.value ? createVNode(TDateFooter, {
        "presets": props.presets,
        "onPresetClick": props.onPresetClick,
        "enableTimePicker": props.enableTimePicker,
        "onConfirmClick": props.onConfirmClick,
        "presetsPlacement": props.presetsPlacement,
        "selectedValue": props.selectedValue,
        "needConfirm": props.needConfirm
      }, null) : null;
    };
  }
});

export { TExtraContent as default };
